packages/scoutgame-ui/src/providers/DeveloperInfoModalProvider.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+
+import { log } from '@charmverse/core/log';
+import { getDeveloperInfo, type DeveloperInfo } from '@packages/scoutgame/builders/getDeveloperInfo';
+import { useRouter } from 'next/navigation';
+import type { ReactNode } from 'react';
+import { createContext, useCallback, useContext, useMemo, useState } from 'react';
+import { toast } from 'sonner';
+
+import { DeveloperInfoModal } from '../components/common/DeveloperInfoModal/DeveloperInfoModal';
+
+type DeveloperInfoModalContextType = {
+  openModal: (path: string) => void;
+  closeModal: () => void;
+  isLoading: boolean;
+  isOpen: boolean;
+};
+
+const DeveloperInfoModalContext = createContext<DeveloperInfoModalContextType | null>(null);
+
+export function DeveloperInfoModalProvider({ children }: { children: ReactNode }) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [developer, setDeveloper] = useState<DeveloperInfo | null>(null);
+  const router = useRouter();
+
+  const openModal = useCallback((path: string) => {
+    setIsLoading(true);
+    setIsOpen(true);
+    getDeveloperInfo(path)
```

Comment:
Can we cache this with SWR, so that when you open the modal again, it is instant?

Reply:
yeah it will be instant if we do that